pallets/bonds/src/lib.rs

```diff
@@ -2208,32 +2215,35 @@ decl_module! {
             Ok(())
         }
         
-        ///Adding funds to the insurer reserves
-        ///Needs amount transfer implemented
+        ///Adding to the balance of InsurerReserves and freezing Asset Id
         #[weight = 1000]
-        pub fn stake(origin, id: u32, deposit: u32) -> dispatch::DispatchResult {
-            let signer = ensure_signed(origin)?;
-            ensure!(InsurerReserves::<T>::contains_key(&signer, id.clone()), Error::<T>::ReserveNotFound);
+        pub fn stake(origin, id: u32, deposit: u32) -> DispatchResultWithPostInfo {
+            let signer = ensure_signed(origin.clone())?;
+            ensure!(InsurerReserves::<T>::contains_key(signer.clone(), id.clone()), Error::<T>::ReserveNotFound);
+            //Freezing this Asset Id before the change in reserves are made
+            pallet_assets::Module::<T>::freeze_asset(origin.clone(), id.clone())?;
```

Comment:
You cannot freeze the assets completely, you must hold on the reserve only the amount matching of the reserve.

Reply:
correct

Reply:
Also, is a transfer of assets missing in these staking and unstaking functions?

Reply:
no